recipes/magnum-plugins/all/cmake/conan-bugfix-global-target.cmake

```diff
@@ -0,0 +1,24 @@
+# If using 'cmake_find_package', all the components are being added to the global
+# target unconditionally. See generated FindMagnumPlugins.cmake file:
+# 
+# if(NOT ${CMAKE_VERSION} VERSION_LESS "3.0")
+#     if(NOT TARGET MagnumPlugins::MagnumPlugins)
+#         add_library(MagnumPlugins::MagnumPlugins INTERFACE IMPORTED)
+#     endif()
+#     set_property(TARGET MagnumPlugins::MagnumPlugins APPEND PROPERTY
+#                  INTERFACE_LINK_LIBRARIES "${MagnumPlugins_COMPONENTS}")
+# endif()
+# 
+# but it doesn't add the library directories and the linker will fail.
+# 
+# Here we fix this bug (breaking change) for this recipe, we just override
+# the list of targets again.
+# 
+
+
+if(NOT ${CMAKE_VERSION} VERSION_LESS "3.0")
+    if(TARGET MagnumPlugins::MagnumPlugins)
+        set_target_properties(MagnumPlugins::MagnumPlugins PROPERTIES INTERFACE_LINK_LIBRARIES
+            "${MagnumPlugins_MagnumPlugins_LINK_LIBS};${MagnumPlugins_MagnumPlugins_LINKER_FLAGS_LIST}")
```

Comment:
This approach can be ported to `magnum` and it would be interesting for `imgui` too (we should check how the project recommends using the bindings, but they could be injected to the consumer automatically).

This should also be a feature for Conan, better than `cpp_info.srcdirs`.

Reply:
I think this is a feature request to conan, to model that approach elegantly
to be honest, I am personally not a fan of that solution, but many famous projects (`imgui` at least) use it